File: numerical_algos/factor_algo.py
#!/usr/bin/env python3

number = int(raw_input("Enter a number whose factors need to be found: "))
factors = []
stop = int(number**(1/2.0))
current = 3

while number%2 == 0:
	factors.append(2)
	number = int(number/2)
while (current<=stop):
	while number%current == 0:
		factors.append(current)
		number = int(number/current)

	current += 2

if number != 1:
	factors.append(number)

print("Factors for the given number are: ")
for item in factors:
	print(item)

File: numerical_algos/factor_algo.c
#include<stdio.h>
#include<stdint.h>
#include<math.h>
#include<stdlib.h>

int main()
{
	uint32_t arr[100],i=0,number,stop,factor;

	printf("Enter a number whose factors need to be found\n");
	scanf("%u" ,&number);
	
	stop = sqrt(number);
	while(number%2 == 0)
	{
		arr[i]=2;
		number /= 2;
		i++;
	}
	factor = 3;
	while(factor<stop)
	{
		while(number%factor == 0)
		{
			arr[i]=factor;
			number /= factor;
			i++;
		}
		factor = factor + 2;
	}
	if(number != 1)
	{
		arr[i]=number;
		i++;
	}

	stop=i;
	i=0;
	printf("\nFactors are: \n");
	for(i=0;i<stop;i++)
	{
		printf("%u\n" ,arr[i]);
	}
}
	

